### ISSUE TITLE
xrandr multihead setup not working

### ISSUE THREAD

**Issue by jonas-kaiser** [tryone144/compton#7]
Hello,

Switching monitors using xrandr doesn't work correctly when I'm running this compton fork.
I'm running arch on a laptop and when I want to switch to an external monitor using `xrandr --output HDMI1 --primary --mode 2560x1440 --pos 0x0 --output VIRTUAL1 --off --output eDP1 --off` the output breaks.

But running this command from my .xinitrc works.

[screenshot laptop (works)](https://ibb.co/7Cc6BkW)
[screenshot extermal (broken)](https://ibb.co/XV4dB9z)

I posted this Issue in the archlinux forum and someone suggestet this:
> the problem is w/ either your WM or compositor not catching the geometry change (and thus eg. not updating glViewport() for a GL compositor)

When I use the official compton version (from the arch repo) or disable compton everything works fine.

Platform: Arch Linux 5.0.2

GPU, drivers, and screen setup:
Intel Corporation Celeron N3350/Pentium N4200/Atom E3900 Series Integrated Graphics Controller,
xf86-video-intel 1:2.99.917+863+g6afed33b-1 (driver),
Laptop screen + external Monitor configured with xrandr

[My compton config](http://ix.io/1DAj)

Compton command:
`compton -b --xrender-sync-fence --blur-background --blur-method kawase --blur-strength 10`
or
`compton -b --xrender-sync-fence`

**Comment by tryone144** [comment]
Hi,

What branch are you using? I guess you installed the [AUR package](https://aur.archlinux.org/packages/compton-tryone-git/) based on the older [dual_kawase](https://github.com/tryone144/compton/tree/dual_kawase) branch.
You may try the newer [improved_fbo](https://github.com/tryone144/compton/tree/improved_fbo) branch and see if the issue persists. (The `blur-method` value has been renamed to `dual_kawase`.)

If you kill and restart compton after the xrandr change, is everything displayed correctly again?

**Comment by jonas-kaiser** [comment]
Yea, I was using the AUR package. I installed the improved_fbo version via git now, but the problem is still there.

Killing and restarting compton works.

**Comment by jonas-kaiser** [comment]
I also tried the (original?) [chjj/compton](https://github.com/chjj/compton) now and having the same issue there.

[yshui/compton](https://github.com/yshui/compton) is the only one that works correctly for me so far.

**Comment by tryone144** [comment]
Yeah, as far as I can tell from a quick look, yshui's fork has a “new” callback to handle root window resizes. As my newer branch is still based on his fork before the recent backend splitting, it still isn't handled correctly.

I'm currently trying to make progress with https://github.com/yshui/compton/issues/32 and merging the dual_kawase blur shader into the refactored codebase. This should eventually fix this issue.

**Comment by jonas-kaiser** [comment]
That's nice to hear.

I don't know if this info is useful, but using the xrender backend also works (no nice blur options tho).
I tried with `compton -b --config /dev/null --backend xrender`
